src/NavierStokes.hpp

```diff
@@ -0,0 +1,743 @@
+                           const double &deltat_,
+                           const double &theta_,
+                           const double nu_,
+                           const double p_out_,
+                           const double rho_,
+                           const int case_type_,
+                           const double vel_,
+                           const unsigned int prec_,
+                           const unsigned int use_skew_)
+  : mesh_file_name(mesh_file_name_),
+    degree_velocity(degree_velocity_),
+    degree_pressure(degree_pressure_),
+    T(_T),
+    deltat(deltat_),
+    theta(theta_),
+    nu(nu_),
+    p_out(p_out_),
+    rho(rho_),
+    inlet_velocity(case_type_, vel_),
+    prec(prec_),
+    use_skew(use_skew_),
+    mpi_size(Utilities::MPI::n_mpi_processes(MPI_COMM_WORLD)),
+    mpi_rank(Utilities::MPI::this_mpi_process(MPI_COMM_WORLD)),
+    pcout(std::cout, mpi_rank == 0),
+    mesh(MPI_COMM_WORLD)
+    {
+    }
+
+
+  void
+  setup();
+
+  void
+  solve();
+
+protected:
+
+  // Assemble constant matrices
+  void
+  assemble_constant_matrices();
+
+  // Assemble system 
+  void assemble(const double &time);
+
+  // Solve system
+  void
+  solve_time_step();
+
+  // Output results
+  void
+  output(const unsigned int &time_step) const;
+
+  // Calculate coefficient
+  void
+  calculate_coefficients(double t);
+
+  // Write coefficients on file
+  void
+  write_coefficients_on_files();
+  
+  // Mesh file
+  const std::string mesh_file_name;
+ 
+  // Polynomial degree
+  const unsigned int degree_velocity;
+  const unsigned int degree_pressure;
+
+  // Total Time 
+  double T;
+
+  // Time step
+  double deltat;
+
+  // Theta parameter of the theta method.
+  double theta;
+
+  // Kinematic viscosity [m2/s]
+  double nu = 0.001;   
+
+  // Outlet pressure [Pa]
+  double p_out = 10.0;
+
+  // density [Kg/m^3]
+  double rho = 1.0; 
+
+  // Inlet velocity
+  InletVelocity inlet_velocity;
+
+  int prec = 0; // 0->diagonal, 1->SIMPLE, 2->ASIMPLE, 2->yosida
+
+  int use_skew = 0; 
+
+  // MPI processes
+  const unsigned int mpi_size;
+
+  // Current MPI process
+  const unsigned int mpi_rank;
+
+  // Parallel output stream
+  ConditionalOStream pcout;
+
+//----------------------------------------------------------------------------
+  
+  // Cylinder diameter, fixed parameter
+  static constexpr double cylinder_diameter = 0.1;
+
+  // Cylinder height, fixed parameter
+  static constexpr double cylinder_height = 0.41;
+
+  // Forcing term
+  ForcingTerm forcing_term;
+
+  // Time
+  double time;
+
+  // Initial Condition
+  FunctionU0 u_0;
+
+  // function g
+  FunctionG function_g;
+
+  // function h
+  FunctionH function_h;
+
+  // Vector of all the drag coefficients
+  std::vector<double> drag_coefficients; 
+  
+  // Vector of all the drag coefficients
+  std::vector<double> lift_coefficients;
+
+  // Retrive Drag/Lift coefficient multiplicative constant
+  double get_drag_lift_multiplicative_const(double t) {
```

Comment:
```suggestion
  double get_drag_lift_multiplicative_const(double t) const {
```
If a method does not require to change the state of a class, always declare it `const`, so that it can also be called on `const` instances of (or `const` references to) that class.